crates/cdk/src/mint.rs

```diff
@@ -328,23 +321,24 @@ impl Mint {
             ..
         } = blinded_message;
 
-        let keyset = self
+        let keyset_info = self
             .localstore
             .get_keyset(keyset_id)
             .await?
             .ok_or(Error::UnknownKeySet)?;
 
         let active = self
             .localstore
-            .get_active_keyset_id(&keyset.unit)
+            .get_active_keyset_id(&keyset_info.unit)
             .await?
             .ok_or(Error::InactiveKeyset)?;
 
         // Check that the keyset is active and should be used to sign
-        if keyset.id.ne(&active) {
+        if keyset_info.id.ne(&active) {
             return Err(Error::InactiveKeyset);
         }
 
+        let keyset = self.generate_keyset(keyset_info.clone());
```

Comment:
I don't think we should regenerate the full `keyset` for every blind sign, it would negatively effect performance. Maybe we can keep the active `keysets` in memory but not persisted to the db. This way the active `keysets` can be generated from the seed only once on mint start up and not on each mint and swap request.

Reply:
Yes, will do. I wasn't sure how invasive this change should be.